server/controllers/signup.js

```diff
@@ -0,0 +1,54 @@
+const bcrypt = require("bcrypt");
+const Boom = require("@hapi/boom");
+
+const signupSchema = require("../utils/validation");
+const getUserByEmail = require("../database/queries/getEmail");
+const signupUser = require("../database/queries/signup");
+const getUserActivity = require("../database/queries/getUserActivity");
+const calculateDailyCalories = require("../utils/dailyCalories");
+
+const signup = async (req, res, next) => {
+  try {
+    const userData = req.body;
+    const { activity_id, email,password} = userData;
+    try {
+      await signupSchema.validateAsync(userData, { abortEarly: false });
+    } catch (err) {
+      throw Boom.badRequest(
```

Comment:
As hassa said before this can be done with badData() instead of badRequest since this just means that the request was successfully sent but the data in it is incorrect